src/hotspot/share/utilities/stringUtils.cpp

```diff
@@ -142,3 +143,22 @@ ccstrlist StringUtils::CommaSeparatedStringIterator::canonicalize(ccstrlist opti
   canonicalized_list[i] = '\0';
   return canonicalized_list;
 }
+
+char* StringUtils::truncate_middle(const char* s, char* out, size_t outlen) {
+  precond(out != nullptr && s != nullptr && outlen > 0);
+  constexpr int dots = 2;
+  const int maxchars = checked_cast<int>(outlen) - 1;
+  const int l = checked_cast<int>(strlen(s));
+  // Impose some reasonable length below which we just truncate dumbly (4 chars each for head/tail)
+  constexpr int smart_truncation_threshold = dots + (4 * 2);
```

Comment:
I'm really not understanding the "smart-threshold" here; nor does this seem worthy of being a utility function.

Reply:
> I'm really not understanding the "smart-threshold" here; nor does this seem worthy of being a utility function.

I think it is a good string utility function; at least it is not only related only to the ASAN support but could be used for other things.